### PR TITLE
added support for getting all friends of the user

### PR DESCRIPTION
**Pull request by recurse-id**

### CODE REVIEW

**Comment by magni-** [comment]
Just a heads up: this doesn't necessarily return all friends, you may have to paginate. See https://developer.venmo.com/docs/endpoints/users

**Comment by urvishparikh** [comment]
without providing the before, after or limit parameters how many friends does the GET request respond with?

**Comment by magni-** [comment]
The default limit is 20 users.

**Comment by eliperkins** [comment]
Thanks for submitting this, @urvishparikh ! :octocat: 

What @magni- mentioned is correct, the API does not return all users at once. Would you mind refactoring the method to be `+- (void)getFriendsWithLimit:(NSNumber *)limit beforeUserID:(NSString *)beforeUserID afterUserID:(NSString *)afterUserID completionHandler:(VENGenericRequestCompletionHandler)handler;`? You can find more documentation of the request scheme and API at the link @magni- mentioned (https://developer.venmo.com/docs/endpoints/users). Keep in mind that limit and both user IDs are allowed to be nil. By creating the API this way, we can allow different clients to implement pagination how they see necessary, rather than forcing a style of pagination from within the SDK.

If you'd like help in implementing this, let me know! Cheers.

**Comment by urvishparikh** [comment]
Thanks for the info eliperkins, magni-, I'll have this modification in soon.

**Comment by urvishparikh** [comment]
Is there a test account I can use that has more than 20 friends?

File: venmo-sdk/Venmo.m [hunk 1]
```diff
@@ -298,6 +298,22 @@ - (void)validateAPIRequestWithCompletionHandler:(VENGenericRequestCompletionHand
 }
 
 
+#pragma mark - Friends
+
+- (void) getFriendsWithLimit:(NSNumber *)limit beforeUserID:(NSString *)beforeUserID afterUserID:(NSString *)afterUserID completionHandler:(VENGenericRequestCompletionHandler)handler {
+    
+    [self validateAPIRequestWithCompletionHandler:handler];
+    
+    NSMutableDictionary* parameters = [NSMutableDictionary dictionaryWithDictionary:@{@"access_token": self.session.accessToken, @"before": beforeUserID, @"after": afterUserID, @"limit": limit}];
```
Comment (eliperkins) [inline venmo-sdk/Venmo.m]: This line will crash if the user does not supply `limit`, `beforeUserID`, or `afterUserID`.

Let's add them by checking if they are nil first. Like so:

```objc
if (limit) {
    parameters[@"limit"] = limit
}

if (beforeUserID) {
...
etc
```

Comment (eliperkins) [inline venmo-sdk/Venmo.m]: Additionally, `VENCore` will handle the access token if it is set with `VENCore -setAccessToken:`. See here: https://github.com/venmo/VENCore/blob/211d8fdd9e5489f7d27caa3bf12fb37521223ded/VENCore/Networking/VENHTTP.m#L221-L244

**Comment by eliperkins** [comment]
Unfortunately, Venmo does not have test accounts. I can help you verify though, as my account has more than 20!

I left a couple of pieces of feedback. Let me know if you'd like help testing after you make those changes.

**Comment by urvishparikh** [comment]
Give it a test drive!

**Comment by urvishparikh** [comment]
Eventually we would want to create a completion handler 

`typedef void (^VENGetFriendsCompletionHandler)(VENFriends response, BOOL success, NSError *error);`

VENFriends should have some methods to make a call for the next or previous getFriends request based on pagination.

**Comment by urvishparikh** [comment]
@eliperkins Did it work?

**Comment by TaterSalad** [comment]
@eliperkins where can I access the method mentioned above to get a user's friends list?

+- (void)getFriendsWithLimit:(NSNumber *)limit beforeUserID:(NSString *)beforeUserID afterUserID:(NSString *)afterUserID completionHandler:(VENGenericRequestCompletionHandler)handler;

I can't seem to do that without using the HTTP handler and a GET request

File: venmo-sdk/Venmo.m [hunk 2]
```diff
@@ -298,6 +298,28 @@ - (void)validateAPIRequestWithCompletionHandler:(VENGenericRequestCompletionHand
...
+    NSMutableDictionary *parameters = [[NSMutableDictionary alloc] init];
+    if (limit)
+        parameters[@"limit"] = [limit stringValue];
+    if (beforeUserID)
+        parameters[@"before"] = beforeUserID;
+    if (afterUserID)
+        parameters[@"after"] = afterUserID;
+    
+    [[VENCore defaultCore].httpClient GET:[NSString stringWithFormat:@"users/%@/friends", self.session.user.externalId] parameters:parameters success:^(VENHTTPResponse *response) {
+        handler(response.object, YES, nil);
```
Comment (dasmer) [inline venmo-sdk/Venmo.m]: the object we are passing back should be an array of `VENUser` objects.  Not a dictionary

**Comment by ayanonagon** [comment]
Hi @urvishparikh thanks for this pull-request, and sorry for the delayed response. We’re working on a way better API client for Venmo right now, so we’re no longer actively adding features to this existing SDK. Of course you are more than welcome to add more features in your fork of the project.

Thanks again for your PR and let us know if you have any questions. Thanks! :octocat:

**Pull request closed without merging**